qrexec/policy/parser.py

```diff
@@ -698,12 +718,30 @@ async def execute(self) -> str:
         if request.source == target:
             raise AccessDenied("loopback qrexec connection not supported")
 
-        return f"""\
+        if target in {"@adminvm", "dom0", "uuid:00000000-0000-0000-0000-000000000000"}:
+            return f"""\
+user={self.user or 'DEFAULT'}
+result=allow
+target=@adminvm
+autostart={self.autostart}
+requested_target={request.target}"""
+        elif target.startswith("@dispvm:"):
+            target_info = request.system_info["domains"][target[8:]]
+            return f"""\
 user={self.user or 'DEFAULT'}
 result=allow
-target={self.target}
+target=@dispvm:uuid:{target_info['uuid']}
 autostart={self.autostart}
-requested_target={self.request.target}"""
+requested_target={request.target}"""
+        else:
+            target_info = request.system_info["domains"][target]
+            return f"""\
+user={self.user or 'DEFAULT'}
+result=allow
+target=uuid:{target_info['uuid']}
```

Comment:
Do not change the meaning of the existing field, it's part of the public API (among other things, it's format of `policy.Eval*` services outout). You can add new `target_uuid` field or something like this.